weasyl/middleware.py

```diff
@@ -153,6 +153,48 @@ def cleanup(request):
     return database_session_cleanup_tween
 
 
+def _generate_http2_server_push_headers():
+    """
+    Generates the Link headers to load HTTP/2 Server Push resources which are needed on each pageload. Written
+    as a separate function to only execute this code a single time, since we just need to generate this each
+    time the code is relaunched (e.g., each time the web workers are kicked to a new version of the code).
+
+    A component of ``http2_server_push_tween_factory``
+    :return: A latin-1 encoded string to be loaded into the Link header set inside of ``http2_server_push_tween_factory``
+    """
+    css_resource_preload_list = [
+        d.get_resource_path('css/site.css'),
+        d.cdnify_url('/static/fonts/museo500.css'),
+    ]
+    css_preload = ", ".join('<' + item + '>; rel=preload; as=style' for item in css_resource_preload_list)
+
+    js_resource_preload_list = [
+        d.cdnify_url('/static/jquery-2.2.4.min.js'),
+        d.cdnify_url('/static/typeahead.bundle.min.js'),
+        d.cdnify_url('/static/marked.js?' + d.CURRENT_SHA),
+        d.cdnify_url('/static/scripts.js?' + d.CURRENT_SHA),
+    ]
+    js_preload = ", ".join('<' + item + '>; rel=preload; as=script' for item in js_resource_preload_list)
+    return ", ".join([js_preload, css_preload]).encode('latin-1')
```

Comment:
Time for my usual minor and unimportant change requests! /o\ Since there should never be any non-ASCII in here, can you encode as ASCII instead of Latin-1? Also, `js_preload` and `css_preload` could be lists with `", ".join(js_preload + css_preload)`. (The lists are never going to be empty, so all it does is DRY the comma, but…)